src/app/components/Modals/UnbondWithdrawModal.tsx

```diff
@@ -15,26 +16,37 @@ export const MODE_WITHDRAW = "withdraw";
 export type MODE = typeof MODE_UNBOND | typeof MODE_WITHDRAW;
 
 interface PreviewModalProps {
-  delegationHeight: number;
   open: boolean;
   onClose: (value: boolean) => void;
   onProceed: () => void;
   mode: MODE;
   awaitingWalletResponse: boolean;
+  delegationsAPI: DelegationInterface[];
+  txID: string;
 }
 
 export const UnbondWithdrawModal: React.FC<PreviewModalProps> = ({
-  delegationHeight,
   open,
   onClose,
   onProceed,
   mode,
   awaitingWalletResponse,
+  delegationsAPI,
+  txID,
 }) => {
   const { coinName, networkName } = getNetworkConfig();
   const versionInfo = useVersionInfo();
 
-  const globalParams = versionInfo?.currentVersion;
+  const delegation = delegationsAPI.find(
+    (delegation) => delegation.stakingTxHashHex === txID,
+  );
```

Comment:
Wrap up any computation logic in useMemo inside render method
```suggestion
  const delegation = useMemo(() => delegationsAPI.find(
    (delegation) => delegation.stakingTxHashHex === txID,
  ), [delegationsAPI, txID]);
```